### ISSUE TITLE
PLR0913 excludes kwargs

### ISSUE THREAD

**Issue by trag1c** [charliermarsh/ruff#4037]
...unless this is expected behavior (both Ruff and Pylint's docs never specify if this check is for positional args only).

```py
def f(a, b, c, d, e, *, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t):
    ...


def g(a, b, c, d, e, f):
    ...
```
```
test.py:5:5: PLR0913 Too many arguments to function call (6 > 5)
```

If this is to be expected then I think it would be good to specify this rule is posargs only :)

**Comment by charliermarsh** [comment]
In testing, it seems like Pylint has the same behavior. I wonder if it's intentional.

**Comment by charliermarsh** [comment]
I think we should change it to include positional-only and keyword-only arguments.

**Comment by trag1c** [comment]
In that case I'd consider bumping the limit a bit.

**Comment by dhruvmanila** [comment]
You can configure the limit using https://beta.ruff.rs/docs/settings/#max-args

```toml
[tool.ruff.pylint]
max-args = 5
```

**Comment by trag1c** [comment]
I was referring to the default 😅

**Comment by dhruvmanila** [comment]
> I was referring to the default 😅

Oh ok, my bad :)

> I think we should change it to include positional-only and keyword-only arguments.

Do you mean to split the rule into positional-only and keyword-only or include both types in the same rule?

**Comment by charliermarsh** [comment]
Ah yeah, I was suggesting that we retain a single rule, and just change the logic to include positional-only and keyword-only arguments in the "argument count" total.

I would probably err on the side of retaining the default, as long as its the same as Pylint's default, since _most_ functions don't have keyword- or positional-only arguments, and so in most cases we'll retain compatibility. What do you think?

**Comment by dhruvmanila** [comment]
If you're going for compatibility, then yes the total count should be good to maintain it in most cases. But, in a few rules we've diverged from the default implementation in terms of avoiding false positive/negative.

We could also open this issue in Pylint to make them aware, it might be that this was unintended from their side as well.

**Comment by charliermarsh** [comment]
Created an issue here: https://github.com/pylint-dev/pylint/issues/8667

**Comment by basnijholt** [comment]
I much preferred the old implementation.

5 arguments seems unrealistically low.

Perhaps we should have a rule that implements the old behavior? :-)

**Comment by JonathanPlasse** [comment]
https://beta.ruff.rs/docs/settings/#max-args
